mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java

```diff
@@ -194,13 +195,24 @@ private boolean isInline(Optional<ContentDispositionField> contentDispositionFie
     }
 
     private boolean isAttachment(Entity part, Context context) {
-        if (context == Context.BODY && isTextPart(part)) {
+        boolean bodyTextCriterion = context == Context.BODY && isTextPart(part);
+
+        boolean attachmentDispositionCriterion = getContentDispositionField(part)
+            .map(ContentDispositionField::getDispositionType)
+            .map(dispositionType -> dispositionType.toLowerCase(Locale.US))
+            .map(ATTACHMENT_CONTENT_DISPOSITIONS::contains)
+            .orElse(false);
+
+        boolean attachmentContentTypeCriterion =  getContentTypeField(part)
+            .map(ContentTypeField::getMimeType)
+            .map(dispositionType -> dispositionType.toLowerCase(Locale.US))
+            .map(ATTACHMENT_CONTENT_TYPES::contains)
+            .orElse(false);
+
+        if (bodyTextCriterion) {
```

Comment:
what is the point of extracting it to a variable ?

Reply:
I found that it ease reading to extract all conditions affecticting wether or not a part is an attachment to give them a name.

Reply:
you can extract it to a method then, that we you have naming